lexos/models/topword_model.py

```diff
@@ -0,0 +1,377 @@
+                      for row in class_division_map.values]
+
+        # Put groups word count sums into a data frame.
+        group_data = \
+            pd.DataFrame(data=group_sums, index=class_labels, columns=words)
+
+        # Generate analysis result.
+        readable_result = [
+            TopwordModel._z_test_word_list(
+                data_series_one=pd.Series(
+                    data=group_data.loc[group_one_label],
+                    index=words,
+                    name='Class "%s"' % group_one_label),
+                data_series_two=pd.Series(
+                    data=group_data.loc[group_two_label],
+                    index=words,
+                    name='Class "%s"' % group_two_label))
+            for group_one_label, group_two_label in label_combinations]
+
+        return readable_result
+
+    def _get_result(self, class_division_map: pd.DataFrame) -> TopwordResult:
+        """Call the right method corresponding to user's selection.
+
+        :return: a namedtuple that holds the topword result, which contains a
+                 header and a list of pandas series."""
+
+        if self._topword_front_end_option.analysis_option == "allToPara":
+            # Get header and result.
+            header = "Compare Each Document to All the Documents As a Whole"
+            results = self._analyze_file_to_all()
+
+            return TopwordResult(header=header, results=results)
+
+        elif self._topword_front_end_option.analysis_option == "classToPara":
+            # Check if more than one class exists.
+            assert class_division_map.shape[0] > 1, NOT_ENOUGH_CLASSES_MESSAGE
+
+            # Get header and result.
+            header = "Compare Each Document to Other Class(es)"
+            results = self._analyze_class_to_all(
+                class_division_map=class_division_map)
+
+            return TopwordResult(header=header, results=results)
+
+        elif self._topword_front_end_option.analysis_option == "classToClass":
+            # Check if more than one class exists.
+            assert class_division_map.shape[0] > 1, NOT_ENOUGH_CLASSES_MESSAGE
+
+            # Get header and result.
+            header = "Compare a Class to Each Other Class"
+            results = self._analyze_class_to_class(
+                class_division_map=class_division_map)
+
+            return TopwordResult(header=header, results=results)
+
+        else:
+            raise ValueError("Invalid topword analysis option.")
+
+    def get_readable_result(self, class_division_map: pd.DataFrame) -> \
+            TopwordResult:
+        """Gets the readable result to display on the web page.
+
+        :return: a namedtuple that holds the topword result, which contains a
+                 header and a list of pandas series, however it will check the
+                 length of each pandas series and only return the first 20 rows
+                 if the pandas series has length that is longer than 20."""
```

Comment:
I am not sure this is standard way to write ending comment, typically I put `"""` on the next line